pandas/tests/io/test_sql.py

```diff
@@ -2387,12 +2438,14 @@ class _TestMySQLAlchemy:
 
     flavor = "mysql"
     port = 3306
+    future = False
```

Comment:
I think what we usually do to manage two versions of the same library is to add a flag in `pandas.compat`, and sometimes a function to handle both versions behavior, that then we call from our code. Do you think this could be helpful and avoid the test duplication and passing the `future` argument?

Reply:
I think the test duplication with the `future` argument is different from what `pandas.compat` provides.  The `future` argument duplicates the type of sqlalchemy connectables that can be passed to `read_sql_query` and `DataFrame.to_sql`.  Currently, the tests are already duplicated to ensure that both `sqlalchemy.engine.Engine` and `sqlalchemy.engine.Connection` can be passed to these methods, and the fact that these are both allowed also makes the code in `pandas.io.sql` more complicated.  In sqlalchemy 1.4, each of these connectables has a subclass, `sqlalchemy.future.Engine` and `sqlalchemy.future.Connection`.  In `pandas.io.sql`, I did not differentiate between future and non-future, but the test duplication ensures that all 4 of these sqlalchemy connectables work.  Once pandas decides to require sqlalchemy 2.0, the future/non-future duplication will be unnecessary.